server/plugin_release.go

```diff
@@ -175,6 +177,12 @@ func createTag(ctx context.Context, client *GithubClient, owner, repository, tag
 		return errors.Wrap(err, "failed to create tag")
 	}
 
+	if preRelease {
```

Comment:
Unfortunately the workflow here is a bit complicated. 
Once a tag is created, there is an external service that turns that tag into a release. So attempting to mark a release as `pre-release` immediately after creating a tag will not work. We will have to move this code here: https://github.com/mattermost/matterbuild/blob/9222d518dbcf0b4a9658a70083cc7f3417fbed34/server/plugin_release.go#L45

i.e. 
1. Create the tag as normal
2. Right before we start processing the plugin asset on line https://github.com/mattermost/matterbuild/blob/9222d518dbcf0b4a9658a70083cc7f3417fbed34/server/plugin_release.go#L45 mark this release as `pre-release` there.

You will also have to fetch the release by tag using this code here: https://github.com/mattermost/matterbuild/blob/9222d518dbcf0b4a9658a70083cc7f3417fbed34/server/plugin_release.go#L507
I would suggest creating a `getReleaseByTag` func that internally attempts to fetch the release or returns an error on timeout. This func can be used by `markTagAsPreRelease` as well as `getPluginAsset`. 

Let me know if this requires further clarification.

Reply:
@ali-farooq0 Thanks for the pointers. Before moving ahead with this, I was wondering if we can hit the external service to create the `release`, I think it would be great if we can do this because then we can execute the steps/apis according to our dependencies graph. Also curious about what all does this external service do while creating a release.

Reply:
@avinashdhinwa
Creating a release out of a tag is done automatically via the circle/github [config](https://github.com/mattermost/matterbuild/blob/248b2f5aba986cf703128f5750fdca981a48b1bb/.circleci/config.yml#L84). I don't think you have to worry about calling this external service, you can just assume it's either created or wait till it gets created.